### PR TITLE
Add flag to test Postgres servers

### PR DESCRIPTION
**Pull request by msakrejda**

### CODE REVIEW

File: heartbleeder.go [hunk 1]
```diff
@@ -57,3 +72,35 @@ func main() {
...
+
+	binary.BigEndian.PutUint32(message[:4], 8)
+	binary.BigEndian.PutUint32(message[4:], 80877103)
+	_, err = c.Write(message)
+	if err != nil {
+		return nil, fmt.Errorf("could not write to server: %v", err)
+	}
+
+	// read the response
+	var response = make([]byte, 1)
```
Comment (titanous) [inline heartbleeder.go]: Drop the `var`, use `:=`.

File: heartbleeder.go [hunk 2]
```diff
@@ -57,3 +72,35 @@ func main() {
...
+		return nil, fmt.Errorf("could not read server response: %v", err)
+	}
+
+	// if the response is not 'S', no ssl
+	if response[0] != 'S' {
+		return nil, fmt.Errorf("this server does not support SSL")
+	}
+
+	// otherwise, we have a connection to try to heartbeat
+	tlsConf := tls.Config{InsecureSkipVerify: true}
```
Comment (titanous) [inline heartbleeder.go]: This is used above too, could make it a package var.

File: heartbleeder.go [hunk 3]
```diff
@@ -57,3 +72,35 @@ func main() {
...
+func pgStartTLS(addr string) (*tls.Conn, error) {
+	c, err := net.Dial("tcp", addr)
+	if err != nil {
+		return nil, err
+	}
+
+	message := make([]byte, 8)
+
+	binary.BigEndian.PutUint32(message[:4], 8)
+	binary.BigEndian.PutUint32(message[4:], 80877103)
```
Comment (titanous) [inline heartbleeder.go:88]: Add a comment describing these?

File: heartbleeder.go (hunk 2 above)
Comment (msakrejda) [inline heartbleeder.go]: Sure, although the specific TLS config required to test could be different across the two protocols.

Comment (titanous) [inline heartbleeder.go]: Something like this:

``` go
var defaultTLSConfig = tls.Config{InsecureSkipVerify: true}
```

Other protocols can choose not use it if they need more specific info.

**Comment by titanous** [comment]
LGTM. Squash, and I'll merge. Thanks!

**Comment by msakrejda** [comment]
Done. Thank you!

**Pull request merged**